File: frontend/src/services/addressService.js
import api from './api';

const BASE = '/v1/users/me/addresses';

export const addressService = {
  list: () => api.get(BASE),

  create: (body) => api.post(BASE, body),

  update: (addressId, body) => api.put(`${BASE}/${addressId}`, body),

  setDefault: (addressId) => api.patch(`${BASE}/${addressId}/default`, {}),

  remove: (addressId) => api.delete(`${BASE}/${addressId}`),
};

File: backend/user-service/src/main/java/com/demir/ecommerce/userservice/service/AddressService.java
package com.demir.ecommerce.userservice.service;

import com.demir.ecommerce.userservice.dto.address.request.AddressRequest;
import com.demir.ecommerce.userservice.dto.address.response.AddressInternalResponse;
import com.demir.ecommerce.userservice.dto.address.response.AddressResponse;

import java.util.List;

public interface AddressService {

    AddressResponse create(AddressRequest request);

    List<AddressResponse> getMyAddresses();

    AddressResponse update(Long addressId, AddressRequest request);

    void delete(Long addressId);

    AddressResponse setDefault(Long addressId);

    AddressInternalResponse getInternalAddress(Long userId, Long addressId);
}